woudc_data_registry/models.py

```diff
@@ -370,6 +376,10 @@ def generate_ids(self):
                           for field in self.id_dependencies]
             self.instrument_id = ':'.join(map(str, components))
 
+        # build the deployment id below
+        temp = {self.station_id, self.contributor}
+        self.deployment_id = ':'.join(temp)
```

Comment:
Right setup in `generate_ids`, just be aware that `{item1, item2, item2}` (as opposed to `{key1: value1, key2: value2}` for dictionaries) is Python set notation.

Sets in some Python versions will scramble the order of their items, so you might end up doing `<contributor>:<station_id>`  instead of the `<station_id>:<contributor>` that you want. To prevent this you need to use an ordered data structure like a list.

Also, it would be best to keep with the same style used elsewhere in the file, of creating the list inside of the `join` call (e.g. `':'.join([item1, item2, ...])`.